### PR TITLE
feat: add longhorn/types

### PR DESCRIPTION
**Pull request by FrankYang0529**
#### Which issue(s) this PR fixes:
https://github.com/longhorn/longhorn/issues/6744

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -76,3 +76,7 @@ require (
 	k8s.io/mount-utils v0.30.0 // indirect
 	k8s.io/utils v0.0.0-20230726121419-3b25d923346b // indirect
 )
+
+replace github.com/longhorn/types => github.com/FrankYang0529/types v0.0.0-20240510025234-5c597a448bbb
```
Comment (Vicente-Cheng) [inline go.mod:80]: Remember to update(remove) this after the corresponding PR merged

File: go.mod [hunk 2]
```diff
@@ -76,3 +76,7 @@ require (
...
+
+replace github.com/longhorn/go-common-libs => github.com/FrankYang0529/go-common-libs v0.0.0-20240510031242-ecfff0cc61d7
```
Comment (Vicente-Cheng) [inline go.mod:82]: ditto

**Review by Vicente-Cheng — DISMISSED** [review]
Just a reminder, others look good to me!

**Comment by innobead** [comment]
@FrankYang0529 needs to update this PR to get it ready.

**Comment by FrankYang0529** [comment]
> @FrankYang0529 needs to update this PR to get it ready.

Rebased it, but need to wait for https://github.com/longhorn/go-common-libs/pull/38.

**Review by innobead — APPROVED** [review]

**Pull request merged**